grafana/resource_cloud_stack.go

```diff
@@ -266,15 +267,18 @@ func ReadStack(ctx context.Context, d *schema.ResourceData, meta interface{}) di
 		return nil
 	}
 
-	FlattenStack(d, stack)
+	err = FlattenStack(d, stack)
+	if err != nil {
+		tflog.Error(ctx, "An error occurred")
+	}
```

Comment:
Instead of logging here, return the error

```suggestion
	if err := FlattenStack(d, stack); err != nil {
		return diag.FromError(err)
	}
```